test/jdk/javax/swing/JToolBar/bug4243930.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 1999, 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/* @test
+ * @bug 4243930
+ * @summary Tests that JToolBar.remove() does not throw StackOverflowError
+ * @run main bug4243930
+ */
+
+import java.awt.event.ActionEvent;
+import java.lang.reflect.InvocationTargetException;
+
+import javax.swing.AbstractAction;
+import javax.swing.JButton;
+import javax.swing.JToolBar;
+import javax.swing.SwingUtilities;
+
+public class bug4243930 {
+
+/** Auxilliary class implementing Action
+ */
+    private static class NullAction extends AbstractAction {
+        public void actionPerformed(ActionEvent e){}
+    }
+
+    public static void main(String[] argv) throws InterruptedException,
+            InvocationTargetException {
```

Comment:
```suggestion
    public static void main(String[] argv) throws Exception {
```
This one hasn't been updated.

Reply:
Nice catch. Thanks, updated

Reply:
Thanks again for the review comments and your time @aivanov-jdk . Could you re-review this?